AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/DeleteEmptyDirectory.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (c) 2022 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.servicelayer.scopedstorage
+
+import com.ichi2.anki.model.Directory
+import com.ichi2.compat.CompatHelper
+import com.ichi2.utils.FileUtil.hasFiles
+import timber.log.Timber
+import java.io.FileNotFoundException
+
+data class DeleteEmptyDirectory(val directory: Directory) : MigrateUserData.Operation() {
+    override fun execute(context: MigrateUserData.MigrationContext): List<MigrateUserData.Operation> {
+        if (directory.directory.hasFiles()) {
```

Comment:
looks like this is mid-refactor? either a listFiles !== empty or have to re-add hasFiles

Reply:
:-) I figured as much, no worries